src/sdam/topology.ts

```diff
@@ -435,7 +434,12 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
 
     // connect all known servers, then attempt server selection to connect
     const serverDescriptions = Array.from(this.s.description.servers.values());
-    connectServers(this, serverDescriptions);
+    this.s.servers = new Map(
+      serverDescriptions.map(serverDescription => [
+        serverDescription.address,
+        createAndConnectServer(this, serverDescription)
```

Comment:
We are doing async actions, but not an awaitable one, the monitor time interval is kicked off to begin sending heartbeats, (or streaming them). I don't believe any code in the remaining steps after this will reference something that's generated by the SDAM machinery. We send subsequent operations through server selection, that's how we await the data that comes out of SDAM. What were you thinking of?